File: solutions/TypeScript/[1009_Easy] Complement of Base 10 Integer/bitwiseComplement.ts
function bitwiseComplement(n: number): number {
  if (n === 0) {
    return 1;
  }

  let num = n;
  const bitLength = Math.floor(Math.log(num) / Math.log(2)) + 1;

  for (let i = 0; i < bitLength; i++) {
    num ^= 1 << i;
  }

  return num;
}

File: solutions/Java/[1009_Easy] Complement of Base 10 Integer/bitwiseComplement.java
class Solution {
  public int bitwiseComplement(int n) {
    if (n == 0) {
      return 1;
    }

    int num = n;
    int bitLength = (int) (Math.log(num) / Math.log(2)) + 1;

    for (int i = 0; i < bitLength; i++) {
      num ^= 1 << i;
    }

    return num;
  }
}
